btcutil/psbt/partial_output.go

```diff
@@ -144,8 +146,23 @@ func (po *POutput) deserialize(r io.Reader) error {
 			)
 
 		default:
-			// Unknown type is allowed for inputs but not outputs.
-			return ErrInvalidPsbtFormat
```

Comment:
Was this a BIP level thing? Or just a simplification that was made during the initial implementation as we didn't have a use case at that time?

Reply:
I think it was just forgotten or ignored during the initial implementation. I added a unit as a last commit to make sure encoding/decoding of the unknown fields works across all levels.